project/src/main/java/com/google/sps/data/RestaurantScorer.java

```diff
@@ -0,0 +1,31 @@
+package com.google.sps.data;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public final class RestaurantScorer {
+  /** Class should not be instantiated. */
+  private RestaurantScorer() {}
+
+  public static double score(Restaurant restaurant, JSONObject preferences) throws JSONException {
+    int preferredPriceLevel = preferences.getJSONObject("priceLevel").getInt("pref");
+    String preferredDiningExp = preferences.getJSONObject("diningExp").getString("pref");
+    int priceLevelWeight = preferences.getJSONObject("priceLevel").getInt("weight");
+    int diningExpWeight = preferences.getJSONObject("diningExp").getInt("weight");
+    double score = 0;
+    double restaurantRating = restaurant.getAvgRating();
+    if (restaurant.getPriceLevel() == preferredPriceLevel) {
+      score += priceLevelWeight;
+    }
+    if (restaurant.getPlaceTypes().contains(preferredDiningExp)) {
+      score += diningExpWeight;
+    }
+    if (restaurantRating >= 3) {
+      score += restaurantRating / 5;
+    } else if (restaurantRating > 0) {
+      // Subtract more points from score for lower rating. This calculation will be improved.
+      score -= (0.6 - restaurantRating / 5);
```

Comment:
Can this just be an else (instead of else if)? Or can we change this to restaurantRating < 3?  I know that it won't enter the else if block if it's >= 3, but I don't like having overlapping cases here (i.e. if it's >= 3, it also is > 0).

Reply:
restaurantRating is set to -1 if there is no rating in the data that comes back from the API, so I had this to make sure that it only considers ratings that exist in the data, but maybe I can change the default value for the rating to something different that would work here?

Reply:
how about : 
boolean hasRating = restaurantRating != -1;
if (hasRating) {
   if (rating >= avg_rating) {
     score = ...
   } else {
     score = ... 
   }
}

its a little bit more nesting so optionally to avoid that, you can also return early: 
if (!hasRating): return score/maxPoints;
 if (rating >= avg_rating) {
     score = ...
 } else {
     score = ... 
 }

Reply:
I changed it to only have one if statement (if rating > 0) in commit 5033396 that bascially does the same thing (except adds slightly less points for a good rating)... let me know if this looks better. I can also change what I have in that commit to the hasRating way that Surbhi wrote above since that suggestion seems clearer to read in my opinion.